web_generator/lib/src/interop_gen/transform/transformer.dart

```diff
@@ -23,17 +23,29 @@ class Transformer {
   /// The type checker for the given program
   final ts.TSTypeChecker typeChecker;
 
-  /// A set of declarations to export
+  /// A set of declarations to export updated during transformation
   final Set<String> exportSet;
 
+  /// A set of declarations to filter for
+  final List<String> filterDeclSet;
+
+  /// The declarations as globs
+  List<RegExp> get filterDeclSetPatterns => filterDeclSet.map((decl) {
+        final escapedDecl = RegExp.escape(decl);
+        if (escapedDecl == decl) return RegExp('^$decl\$');
+        return RegExp(decl);
+      }).toList();
+
   /// namer, for giving elements unique names
   final UniqueNamer namer;
 
   final ProgramDeclarationMap programMap;
 
   Transformer(this.programMap, this.typeChecker,
-      {Iterable<String> exportSet = const []})
+      {Iterable<String> exportSet = const [],
```

Comment:
nit: If we're always converting these iterables to a subtype, expect the subtype instead e.g. `Set<String> exportSet` and `List<String> filterDeclSet`. This avoids the case where we unnecessarily call e.g. `toList` on a `List`.